core/src/parse.rs

```diff
@@ -324,35 +327,54 @@ fn line(input: RawSpan) -> ParseResult<'_, Option<SpanNode<Statement<Span>>>> {
     )(input)
 }
 
-fn parse(input: &str) -> Result<Program<Span>, CompileError> {
+fn parse(
+    input: &str,
+) -> Result<Program<Span>, Vec<SourceErrorWrapper<CompileError>>> {
     match Program::parse(RawSpan::new(input)) {
         Ok((_, program)) => Ok(program),
         Err(nom::Err::Error(e)) | Err(nom::Err::Failure(e)) => {
-            match e.errors.as_slice() {
-                [(substring, VerboseErrorKind::Nom(ErrorKind::Eof)), ..] => {
-                    // If the error is EOF that means there was remaining
-                    // input that was not parsed
-                    // so they put in a bad keyword
```

Comment:
Is this comment not relevant anymore?

Reply:
Technically it still is now, but I got rid of it while I was trying to port this code. I'm planning on cleaning up the parsing errors soon, and I'm hoping we don't need to handle this case like this anymore.